src/main/java/models/ShoppingCart.java

```diff
@@ -10,12 +10,14 @@
 @Entity
 public class ShoppingCart {
     private Long id;
-    @JsonIgnore
     private Set<Book> books;
     @JsonIgnore
     private Customer customer;
 
     public ShoppingCart(){ }
+    public ShoppingCart(Customer customer){
```

Comment:
I think we should follow the convention of setting the entity relationships via getters, since that is done everywhere else across the model. This is a non functional comment.

Reply:
Agreed.
Addressed in 16afba54f8c216f7beb7ebb1cd71e9c42229a205